modules/tinymce/src/core/main/ts/api/Options.ts

```diff
@@ -1035,118 +1043,120 @@ const isDisabled = option('disabled');
 const getExtendedMathmlAttributes = option('extended_mathml_attributes');
 const getExtendedMathmlElements = option('extended_mathml_elements');
 const shouldIndentOnTab = option('lists_indent_on_tab');
+const getListMaxDepth = (editor: Editor): Optional<number> =>
+  Optional.from(editor.options.get('list_max_depth'));
```

Comment:
You might notice you had to import `Optional` in order to make this work. We don't use optionals in core settings.

Remove the optional, and just use `Type.isNumber` in the code that reads the option.

Reply:
Is there a particular reason why it's all internal? It it becomes cause we seldom add core options and until now not needed an option.

Reply:
Hard to say. The vast majority of these have their defaults encoded in this file, so they don't need to be options.

Other cases use things like `Type.isFunction()` to determine whether to perform an action instead of making it an optional value with `.each()`. But those would all have been written before we had option validation.

Maybe we side-step the question and make the default `-1` indicating no max depth 😂 

If we were going to start returning optional values I'd want to create a `wrapOptional` function as we do on some of the premium features.

Reply:
I assumed we wanted to avoid not wrapping into Optionals? Not doing it just seems like I'd have to do it where we call it anyway, so why not do it here?

Reply:
If we don't have a reason not to have it I think it's better to keep it? I don't see a point to have all calls wrap it in an optional?